starcheck/star_probs.py

```diff
@@ -0,0 +1,221 @@
+        t_ccd = warm_t_ccd
+
+    elif n_acq_above_min(cold_t_ccd) <= 0:
+        # If there are not enough ACQ stars at the coldest CCD temperature then stop there
+        # as well.  The ACA thermal model will never predict a temperature below this
+        # value so this catalog will fail thermal check.
+        t_ccd = cold_t_ccd
+
+    else:
+        # At this point there must be a zero in the range [cold_t_ccd, warm_t_ccd]
+        t_ccd = brentq(n_acq_above_min, cold_t_ccd, warm_t_ccd, xtol=1e-4, rtol=1e-4)
+
+    n_acq = n_acq_above_min(t_ccd) + min_n_acq
+
+    return t_ccd, n_acq
+
+
+@jit(nopython=True)
+def _prob_n_acq(star_probs, n_stars, n_acq_probs):
+    """
+    Jit-able portion of prob_n_acq
+    """
+    for cfg in range(1 << n_stars):
+        prob = 1.0
+        n_acq = 0
+        for slot in range(n_stars):
+            success = cfg & (1 << slot)
+            if success > 0:
+                prob = prob * star_probs[slot]
+                n_acq += 1
+            else:
+                prob = prob * (1 - star_probs[slot])
+        n_acq_probs[n_acq] += prob
+
+
+def prob_n_acq(star_probs):
+    """
+    Given an input array of star acquisition probabilities ``star_probs``,
+    return the probabilities of acquiring exactly n_acq stars, where n_acq
+    is evaluated at values 0 to n_stars.  This is returned as an array
+    of length n_stars.  In addition the cumulative sum, which represents
+    the probability of acquiring n_acq or fewer stars, is returned.
+
+    :param star_probs: array of star acq probabilities (list or ndarray)
+
+    :returns n_acq_probs, cum_n_acq_probs: tuple of ndarray, ndarray
+    """
+    star_probs = np.array(star_probs, dtype=np.float64)
+    n_stars = len(star_probs)
+    n_acq_probs = np.zeros(n_stars + 1, dtype=np.float64)
+
+    _prob_n_acq(star_probs, n_stars, n_acq_probs)
+
+    return n_acq_probs, np.cumsum(n_acq_probs)
+
+
+def acq_success_prob(date=None, t_ccd=-19.0, mag=10.0, color=0.6, spoiler=False):
+    """
+    Return probability of acquisition success for given date, temperature and mag.
+
+    Any of the inputs can be scalars or arrays, with the output being the result of
+    the broadcasted dimension of the inputs.
+
+    This is based on the dark model and acquisition success model presented
+    in the State of the ACA 2013, and subsequently updated to use a Probit
+    transform and separately fit B-V=1.5 stars.  This is available in the
+    state_of_aca repo as fit_sota_model_probit.ipynb.
+
+    :param date: Date(s) (scalar or np.ndarray, default=NOW)
+    :param t_ccd: CD temperature(s) (degC, scalar or np.ndarray, default=-19C)
+    :param mag: Star magnitude(s) (scalar or np.ndarray, default=10.0)
+    :param color: Star color(s) (scalar or np.ndarray, default=0.6)
+    :param spoil: Star spoiled (boolean or np.ndarray, default=False)
+
+    :returns: Acquisition success probability(s)
+    """
+    from mica.archive.aca_dark.dark_model import get_warm_fracs
```

Comment:
This is a perfect example of why localized imports are dangerous, and how I need to follow my own advice!  Before the star working group I had scanned the imports for this file (at the top) and had made a mental note that there were no real dependencies that don't have Matlab equivalents.  WRONG.